code/modules/religion/honorbound/honorbound_trauma.dm

```diff
@@ -25,43 +25,77 @@
 	RegisterSignal(owner, COMSIG_ATOM_WAS_ATTACKED, PROC_REF(on_attacked))
 
 	//signal that checks for dishonorable attacks
-	RegisterSignal(owner, COMSIG_MOB_CLICKON, PROC_REF(attack_honor))
+	RegisterSignal(owner, COMSIG_LIVING_EARLY_UNARMED_ATTACK, PROC_REF(unarmed_attack_honor))
+	RegisterSignal(owner, COMSIG_MOB_ITEM_ATTACK, PROC_REF(attack_honor))
+	RegisterSignal(owner, COMSIG_MOB_TRYING_TO_FIRE_GUN, PROC_REF(fire_gun_honor))
 	var/datum/action/cooldown/spell/pointed/declare_evil/declare = new(owner)
 	declare.Grant(owner)
 	return ..()
 
 /datum/brain_trauma/special/honorbound/on_lose(silent)
 	owner.clear_mood_event("honorbound")
 	UnregisterSignal(owner, list(
-		COMSIG_ATOM_WAS_ATTACKED,
-		COMSIG_MOB_CLICKON,
+		COMSIG_LIVING_EARLY_UNARMED_ATTACK,
+		COMSIG_MOB_ITEM_ATTACK,
+		COMSIG_MOB_TRYING_TO_FIRE_GUN,
 		COMSIG_MOB_CAST_SPELL,
 		COMSIG_MOB_FIRED_GUN,
 	))
 	return ..()
 
-/// Signal to see if the trauma allows us to attack a target
-/datum/brain_trauma/special/honorbound/proc/attack_honor(mob/living/carbon/human/honorbound, atom/clickingon, list/modifiers)
+/datum/brain_trauma/special/honorbound/proc/unarmed_attack_honor(mob/living/carbon/human/honorbound, atom/target, proximity_flag, modifiers)
 	SIGNAL_HANDLER
 
 	if(modifiers[ALT_CLICK] || modifiers[SHIFT_CLICK] || modifiers[CTRL_CLICK] || modifiers[MIDDLE_CLICK])
 		return
-	if(!isliving(clickingon))
+
+	if(!proximity_flag || !isliving(target))
+		return NONE
+
+	var/mob/living/punched_mob = target
+
+	if(!(punched_mob in guilty))
+		check_visible_guilt(punched_mob)
+
+	if((honorbound.combat_mode || modifiers[RIGHT_CLICK]) && !is_honorable(honorbound, punched_mob))
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/// Signal to see if the trauma allows us to attack a target with a weapon
+/datum/brain_trauma/special/honorbound/proc/attack_honor(mob/living/honorbound, atom/attacked, params)
+	SIGNAL_HANDLER
+
+	if(!isliving(attacked))
 		return
 
-	var/mob/living/clicked_mob = clickingon
+	var/mob/living/attacked_mob = attacked
 	var/obj/item/weapon = honorbound.get_active_held_item()
 
-	if(!honorbound.DirectAccess(clicked_mob) && !isgun(weapon))
-		return
 	if(weapon?.item_flags & NOBLUDGEON)
 		return
-	if(!honorbound.combat_mode && (HAS_TRAIT(clicked_mob, TRAIT_ALLOWED_HONORBOUND_ATTACK) || ((!weapon || !weapon.force) && !LAZYACCESS(modifiers, RIGHT_CLICK))))
+
+	if(!(attacked_mob in guilty))
+		check_visible_guilt(attacked_mob)
+
+	if((weapon?.item_flags & NEEDS_PERMIT || honorbound.combat_mode || weapon?.force > 0) && !is_honorable(honorbound, attacked_mob))
```

Comment:
```suggestion
	if(((weapon?.item_flags & NEEDS_PERMIT) || honorbound.combat_mode || weapon?.force > 0) && !is_honorable(honorbound, attacked_mob))
```
I think checking needs permit is a lil odd but it's w/e

Reply:
I did think it was a weird name because mostly it's used to indicate that the item is somehow dangerous or a weapon of some sort. Even abstract objects having it is a bit strange. I reckon we could probably come back to it later.